GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -163,34 +164,44 @@ public async Task ReloadAsync()
                 treeMain.BeginUpdate();
                 _rootNodes.ForEach(t => t.IgnoreSelectionChangedEvent = true);
                 var tasks = _rootNodes.Select(r => r.ReloadAsync(token)).ToArray();
-                await Task.WhenAll(tasks).ConfigureAwait(false);
+                await Task.WhenAll(tasks).ConfigureAwait(true);
             }
             finally
             {
-                await this.SwitchToMainThreadAsync();
+                ThreadHelper.AssertOnUIThread();
+
                 if (!token.IsCancellationRequested)
                 {
                     Enabled = true;
+                }
+
+                // Need to end the update for the selected node to scroll into view below under certain conditions
+                // (e.g. when the treeview gets larger, trying to make the selected node visible doesn't work).
+                treeMain.EndUpdate();
+
+                if (!token.IsCancellationRequested)
+                {
                     var selectedNode = treeMain.AllNodes().FirstOrDefault(n =>
                         _rootNodes.Any(rn => $"{rn.TreeViewNode.FullPath}{treeMain.PathSeparator}{currentBranch}" == n.FullPath));
                     if (selectedNode != null)
                     {
-                        treeMain.SelectedNode = selectedNode;
-                        ExpandPathToSelectedNode();
+                        SetSelectedNode(selectedNode);
                     }
 
                     _rootNodes.ForEach(t => t.IgnoreSelectionChangedEvent = false);
                 }
-
-                treeMain.EndUpdate();
             }
+        }
 
-            void ExpandPathToSelectedNode()
-            {
-                treeMain.Scrollable = false; // disable scrolling, so the next call does not horizontally (nor vertically) scroll
-                treeMain.SelectedNode.EnsureVisible();
-                treeMain.Scrollable = true;
-            }
+        private void SetSelectedNode(TreeNode node)
```

Comment:
Why not a local function?

Reply:
Personally I think we overuse private functions, more often than not they make the code significantly harder to test.